### ISSUE TITLE
does not work with `s/or` and multiple spec maps

### ISSUE THREAD

**Issue by bmabey** [bhb/expound#64]
I'm running into this assertion error:
```clojure
(defn key->spec [keys problems]
  (assert (apply = (map :expound/via problems)))
....
```
https://github.com/bhb/expound/blob/master/src/expound/printer.cljc#L47

Any idea why this may be?
I'll dig in and see what the actual values are. If anything I think the error message here could be improved since the vanilla assert message is not helpful.

Here is the full stack trace:
```
1. Unhandled java.lang.AssertionError
   Assert failed: (apply = (map :expound/via problems))

              printer.cljc:   47  expound.printer$key__GT_spec/invokeStatic
              printer.cljc:   46  expound.printer$key__GT_spec/invoke
              printer.cljc:  186  expound.printer$print_spec_keys$fn__21784/invoke
              printer.cljc:  186  expound.printer$print_spec_keys/invokeStatic
              printer.cljc:  177  expound.printer$print_spec_keys/invoke
                alpha.cljc:  225  expound.alpha$explain_missing_keys/invokeStatic
                alpha.cljc:  216  expound.alpha$explain_missing_keys/invoke
                alpha.cljc:  230  expound.alpha$eval21950$fn__21951/invoke
              MultiFn.java:  260  clojure.lang.MultiFn/invoke
                alpha.cljc:  242  expound.alpha$eval21954$fn__21955/invoke
              MultiFn.java:  260  clojure.lang.MultiFn/invoke
                alpha.cljc:  459  expound.alpha$problem_group_str1/invokeStatic
                alpha.cljc:  457  expound.alpha$problem_group_str1/invoke
                alpha.cljc:  504  expound.alpha$printer_str$iter__22065__22069$fn__22070/invoke
              LazySeq.java:   40  clojure.lang.LazySeq/sval
              LazySeq.java:   49  clojure.lang.LazySeq/seq
                 Cons.java:   39  clojure.lang.Cons/next
              LazySeq.java:   81  clojure.lang.LazySeq/next
                   RT.java:  703  clojure.lang.RT/next
                  core.clj:   64  clojure.core/next
                string.clj:  180  clojure.string/join
                string.clj:  180  clojure.string/join
                alpha.cljc:  503  expound.alpha$printer_str/invokeStatic
                alpha.cljc:  479  expound.alpha$printer_str/invoke
                alpha.cljc:  515  expound.alpha$custom_printer$fn__22092/invoke
                alpha.cljc:  520  expound.alpha$printer/invokeStatic
                alpha.cljc:  517  expound.alpha$printer/invoke
                 alpha.clj:  250  clojure.spec.alpha/explain-out
                 alpha.clj:  246  clojure.spec.alpha/explain-out
                  test.clj:  115  orchestra.spec.test/spec-checking-fn/conform!/fn
                  test.clj:  115  orchestra.spec.test/spec-checking-fn/conform!
                  test.clj:  125  orchestra.spec.test/spec-checking-fn/fn
               RestFn.java:  137  clojure.lang.RestFn/applyTo
                  Var.java:  702  clojure.lang.Var/applyTo
                  core.clj:  659  clojure.core/apply
                  core.clj:  652  clojure.core/apply
```

**Comment by bhb** [comment]
@bmabey Thanks very much for reporting this!

This is an internal check that unfortunately cannot be resolved by your client code - it means that Expound incorrectly assumes that all problems refer to the same piece of invalid data. If that's *not* true, Expound is going to have a bug anyway, so this assert catches the problem early. That's why the error message isn't too good - the only resolution is to report the bug, rather than change your code 😄 . 

> I think the error message here could be improved since the vanilla assert message is not helpful.

That's a good idea! I could at least add a message asking users to report the bug, so they don't wonder how to resolve it.

All that said, you've found a bug in Expound. If possible, can you share the code you used to trigger this assert? Thanks!

**Comment by bmabey** [comment]
The bug occurs when using `s/or`. Here is a reduced and contrived example that will trigger it:

```clojure
(s/def :example.place/city string?)
(s/def :example.place/state string?)
(s/def :example/place (s/keys :req-un [:example.place/city :example.place/state]))

(s/def :example.person/first-name string?)
(s/def :example.person/last-name string?)
(s/def :example/person (s/keys :req-un [:example.person/first-name :example.person/last-name]))

(s/def :example/person-or-place  (s/or :person :example/person
                                       :place :example/place))

(s/explain :example/person-or-place {}) ;; works fine
(expound/expound :example/person-or-place {}) ;; blows up
```

**Comment by bhb** [comment]
@bmabey Thanks for the repro! I'll fix ASAP

**Comment by bhb** [comment]
@bmabey Thanks again for reporting this. Can you try `0.4.1-SNAPSHOT` and see if it works for you?

**Comment by bhb** [comment]
@bmabey Does this fix your issue? You can try it out by using `0.4.1-SNAPSHOT`. Thanks!

**Comment by bmabey** [comment]
It is hard for me to reproduce the error in my system now but if the above use case is handled I think this will prevent it from happening again. Thanks!

**Comment by bhb** [comment]
@bmabey No worries, thanks for the feedback!

**Comment by bhb** [comment]
@bmabey I've released Expound 0.5.0, which includes this fix.
